src/main/java/plugily/projects/buildbattle/handlers/themes/vote/VoteMenu.java

```diff
@@ -103,7 +104,14 @@ private NormalFastInv getGUI() {
 
     gui.addCloseHandler(event -> {
       if(arena.getArenaInGameStage() == BaseArena.ArenaInGameStage.THEME_VOTING) {
-        event.getPlayer().openInventory(event.getInventory());
+        plugin.getServer().getScheduler().runTaskLater(plugin, () -> {
+          HumanEntity humanEntity = event.getPlayer();
+          Inventory inventory = event.getInventory();
+
+          if (humanEntity.getOpenInventory().getTopInventory() != inventory) {
+            humanEntity.openInventory(inventory);
+          }
+        }, 3); // Delay operation to make sure the top inventory is closed before opening again
```

Comment:
just runtask without a number, so which is executed a tick later would be enough?

Reply:
I believe there should be a slight delay before this GUI opens. If we instead used the runTask method, which will be called on the next server tick, this GUI will probably not be called because the previous top inventory has not yet closed.

Reply:
I would test it with one server tick and if it doesn't work we can go longer